heron/packing/src/java/com/twitter/heron/packing/PackingUtils.java

```diff
@@ -219,4 +222,142 @@ public static int allocateNewContainer(ArrayList<Container> containers, Resource
+      componentChanges, Resource defaultInstanceResources) {
+    double cpu = 0;
+    long ram = 0;
+    long disk = 0;
+    Map<String, Long> ramMap = TopologyUtils.getComponentRamMapConfig(topology);
+    Map<String, Integer> componentsToScaleDown = PackingUtils.getComponentsToScaleDown(
+        componentChanges);
+    for (String component : componentsToScaleDown.keySet()) {
+      int parallelismChange = -componentChanges.get(component);
+      cpu += parallelismChange * defaultInstanceResources.getCpu();
+      disk += parallelismChange * defaultInstanceResources.getDisk();
+      if (ramMap.containsKey(component)) {
+        ram += parallelismChange * ramMap.get(component);
+      } else {
+        ram += parallelismChange * defaultInstanceResources.getRam();
+      }
+    }
+    return new Resource(cpu, ram, disk);
+  }
+
+  /**
+   * Identifies the resources needed for the components that will be scaled up
+   *
+   * @return Total resources needed
+   */
+  public static Resource getScaleUpResource(TopologyAPI.Topology topology, Map<String, Integer>
+      componentChanges, Resource defaultInstanceResources) {
+    double cpu = 0;
+    long ram = 0;
+    long disk = 0;
+    Map<String, Long> ramMap = TopologyUtils.getComponentRamMapConfig(topology);
+    Map<String, Integer> componentsToScaleUp = PackingUtils.getComponentsToScaleUp(
+        componentChanges);
+    for (String component : componentsToScaleUp.keySet()) {
+      int parallelismChange = componentChanges.get(component);
+      cpu += parallelismChange * defaultInstanceResources.getCpu();
+      disk += parallelismChange * defaultInstanceResources.getDisk();
+      if (ramMap.containsKey(component)) {
+        ram += parallelismChange * ramMap.get(component);
+      } else {
+        ram += parallelismChange * defaultInstanceResources.getRam();
+      }
+    }
+    return new Resource(cpu, ram, disk);
+  }
+
+  /**
+   * Computes the additional resources needed to accommodate a scale up and down operation
+   *
+   * @return Additional resources needed
+   */
+  public static Resource getAdditionalResources(Resource scaleupResource,
+                                                Resource scaledownResource) {
+    double cpuDifference =  scaleupResource.getCpu() - scaledownResource.getCpu();
+    double cpu = cpuDifference < 0 ? 0 : cpuDifference;
+    long ramDifference =  scaleupResource.getRam() - scaledownResource.getRam();
+    long ram =  ramDifference < 0 ? 0 : ramDifference;
+    long diskDifference = scaleupResource.getDisk() - scaledownResource.getDisk();
+    long disk = diskDifference < 0 ? 0 : diskDifference;
+    return new Resource(cpu, ram, disk);
+  }
+
+
+  /**
+   * Computes number of containers needed to accommodate a specific resource request given the container size
+   *
+   * @return Number of containers needed
+   */
+  public static double getRequiredNumContainers(Resource resourceRequest, Resource capacity) {
```

Comment:
@avflor see my other comment re mathematical operations on the object that does them and OO vs helpers.